tests/client/test_client_tenant_token.py

```diff
@@ -0,0 +1,106 @@
+    response = token_client.index('indexUID').search('', {
+        'limit': 5
+    })
+    assert isinstance(response, dict)
+    assert len(response['hits']) == 5
+    assert response['query'] == ''
+
+def test_generate_tenant_token_with_search_rules_on_one_index(get_private_key, empty_index):
+    """Tests create a tenant token with only search rules."""
+    empty_index()
+    empty_index('tenant_token')
+    client = meilisearch.Client(BASE_URL, get_private_key['key'])
+
+    token = client.generate_tenant_token(search_rules=['indexUID'])
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert response['query'] == ''
+    with pytest.raises(MeiliSearchApiError):
+        response = token_client.index('tenant_token').search('')
+
+def test_generate_tenant_token_with_api_key(client, get_private_key, empty_index):
+    """Tests create a tenant token with only search rules."""
+    empty_index()
+    token = client.generate_tenant_token(search_rules=["*"], api_key=get_private_key['key'])
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert response['query'] == ''
+
+def test_generate_tenant_token_with_expires_at(client, get_private_key, empty_index):
+    """Tests create a tenant token with search rules and expiration date."""
+    empty_index()
+    client = meilisearch.Client(BASE_URL, get_private_key['key'])
+    tomorrow = datetime.datetime.now() + datetime.timedelta(days=1)
+
+    token = client.generate_tenant_token(search_rules=["*"], expires_at=tomorrow)
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert response['query'] == ''
+
+def test_generate_tenant_token_with_empty_search_rules_in_list(get_private_key):
+    """Tests create a tenant token without search rules."""
+    client = meilisearch.Client(BASE_URL, get_private_key['key'])
+
+    with pytest.raises(Exception):
+        client.generate_tenant_token(search_rules=[''])
+
+def test_generate_tenant_token_without_search_rules_in_list(get_private_key):
+    """Tests create a tenant token without search rules."""
+    client = meilisearch.Client(BASE_URL, get_private_key['key'])
+
+    with pytest.raises(Exception):
+        client.generate_tenant_token(search_rules=[])
+
+def test_generate_tenant_token_without_search_rules_in_dict(get_private_key):
+    """Tests create a tenant token without search rules."""
+    client = meilisearch.Client(BASE_URL, get_private_key['key'])
+
+    with pytest.raises(Exception):
+        client.generate_tenant_token(search_rules={})
+
+def test_generate_tenant_token_wit_empty_search_rules_in_dict(get_private_key):
+    """Tests create a tenant token without search rules."""
```

Comment:
I noticed the tests has the same description is that ok? :)

Reply:
Good catch I didn't notice this, it is not. Only the last test with the name will be reported.